core/templates/dev/head/components/question-directives/questions-list/questions-list.directive.html

```diff
@@ -40,7 +40,7 @@
   <tr ng-repeat="questionSummary in $ctrl.getQuestionSummaries()"
       class="list-item protractor-test-question-list-item"
       style="height: 4em;"
-      ng-click="$ctrl.editQuestion(questionSummary.summary)"
+      ng-click="$ctrl.editQuestion(questionSummary.summary, questionSummary.skill_difficulties[0])"
```

Comment:
Why is "[0]" singled out? If the zeroth element is special somehow it should be made into a different field.

Reply:
This is because the endpoint from where we get this list can take multiple skill IDs as the parameter and returns difficulties w.r.t each of them, but everywhere we use it in the frontend, we only consider one skill at a time (topic editor has the skill selector and the skill editor itself). So, here only data for that one skill would be there. Hence, why just [0] is used.
Should I modify QuestionsListService to store the skill description and difficulty in a different object which is not an array, since an array is not really needed in the frontend?

Reply:
Yes, in general don't use "0th element of an array" when the array itself isn't needed -- it's misleading conceptually.

Reply:
Done